### PR TITLE
provisioncli: Add install and uninstall actions

### PR DESCRIPTION
**Pull request by jtumber-ibm**
Reuse e2e test code to deploy operator components into existing cluster.

Fixes: #922

### CODE REVIEW

**Review by stevenhorsman — COMMENTED** [review]
Testing these now, but whilst I do @jtumber-ibm - can you add the actions to the doc in https://github.com/jtumber-ibm/cloud-api-adaptor/blob/deploy-cli/test/tools/README.md please. Thanks!

File: test/tools/README.md [hunk 1]
```diff
@@ -28,7 +28,17 @@ export TEST_PROVISION_FILE=${PROPERTIES_FILE_PATH}
...
+
+### Brief Action Explanations
+
+`provision` : Uses the provisioner to create a cluster
+
+`deprovision` : Deletes the cluster that we previously created
+
+`deploy` : Deploy the cloud-api-adaptor using the operator to an existing cluster, must set `KUBECONFIG`
+
+`delete` : Removes the cloud-api-adaptor resources from the cluster, must set `KUBECONFIG`
```
Comment (stevenhorsman) [inline test/tools/README.md:41]: I wonder if something like undeploy/uninstall might be better than delete for this action as you might expect the delete of the cluster to be done under that action?

Comment (stevenhorsman) [reply test/tools/README.md:41]: We use `Uninstall CoCo and cloud-api-adaptor` in the info message, so maybe `uninstall` would be best?

**Review by stevenhorsman — APPROVED** [review]
Apart from maybe a change of the action name and doc clarification, this LGTM. Thanks @jtumber-ibm 

My local testing was: 
- Provision a cluster with the provision action (which also does a deploy)
- Validation the the operator install worked:
```
# kubectl get pods -n confidential-containers-system
NAME                                             READY   STATUS    RESTARTS   AGE
cc-operator-controller-manager-fbb5dcf9d-nfblg   2/2     Running   0          80m
cc-operator-daemon-install-7227r                 1/1     Running   0          79m
cloud-api-adaptor-daemonset-tdmxg                1/1     Running   0          79m
```
- Run the `delete` action and validate that the cloud-api-adaptor and coco operator were uninstalled:
```
# ./caa-provisioner-cli -action=delete
INFO[0000] IamServiceURL is: https://iam.cloud.ibm.com/identity/token.
INFO[0000] VpcServiceURL is: https://eu-gb.iaas.cloud.ibm.com/v1.
INFO[0000] Deleting CoCo operator and cloud-api-adaptor resources
INFO[0000] Uninstall CoCo and cloud-api-adaptor
INFO[0166] Uninstall the controller manager
INFO[0167] Wait for the cc-operator-controller-manager deployment be deleted
# kubectl get pods -n confidential-containers-system
No resources found in confidential-containers-system namespace.
```
- Use the deploy action and check that CoCo and CAA were re-installed:
```
# ./caa-provisioner-cli -action=deploy
INFO[0000] IamServiceURL is: https://iam.cloud.ibm.com/identity/token.
INFO[0000] VpcServiceURL is: https://eu-gb.iaas.cloud.ibm.com/v1.
INFO[0000] Deploying CoCo operator and cloud-api-adaptor resources
INFO[0000] Install the controller manager
Wait for the cc-operator-controller-manager deployment be available
INFO[0016] Customize the overlay yaml file
INFO[0016] Updating caa image tag with fb83fdb36d1899f822b7423729357a9298775968
INFO[0016] Install CoCo and cloud-api-adaptor
Wait for the cc-operator-daemon-install DaemonSet be available
Wait for the pod cc-operator-daemon-install-m5pzb be ready
Wait for the cloud-api-adaptor-daemonset DaemonSet be available
Wait for the pod cloud-api-adaptor-daemonset-5kcx2 be ready
Wait for the kata-remote runtimeclass be created
# kubectl get pods -n confidential-containers-system
NAME                                             READY   STATUS    RESTARTS   AGE
cc-operator-controller-manager-fbb5dcf9d-zt9kg   2/2     Running   0          78s
cc-operator-daemon-install-m5pzb                 1/1     Running   0          62s
cloud-api-adaptor-daemonset-5kcx2                1/1     Running   0          63s
```

File: test/tools/README.md (hunk 1 above)
Comment (jtumber-ibm) [reply test/tools/README.md:41]: I based it off of the `make deploy` and `make delete` already in use. I agree it isn't the best naming though

Comment (jtumber-ibm) [reply test/tools/README.md]: Changed to `install` and `uninstall`

File: test/tools/README.md [hunk 2]
```diff
@@ -28,7 +28,17 @@ export TEST_PROVISION_FILE=${PROPERTIES_FILE_PATH}
 export TEST_PROVISION="yes"
 ./caa-provisioner-cli -actions=${ACTION}
 ```
-`ACTION` supports `provision`, `deprovision` and `uploadimage`.
+`ACTION` supports `provision`, `deprovision`, `install`, `uninstall` and `uploadimage`.
+
+### Brief Action Explanations
+
+`provision` : Uses the provisioner to create a cluster
```
Comment (stevenhorsman) [inline test/tools/README.md:35]: Sorry, I thought I commented this in the last review, but can't spot it now. It's probably worth documenting that `provision` also does the `install` action, not just creates the cluster

**Review by stevenhorsman — APPROVED** [review]
Thanks for the updates. LGTM

**Review by yoheiueda — APPROVED** [review]
LGTM

**Pull request merged**